### PR TITLE
Improve `RecentEra` docs

### PR DESCRIPTION
**Pull request by Anviking**
Related Advice Process Decision: https://docs.google.com/document/d/1xazQNECWex3GWp2-yzAeJGnYvAHchamgNVvFDs3ydcg/edit?tab=t.0#heading=h.gjdgxs

### CODE REVIEW

**Comment by abailly** [comment]
Is this still relevant?

**Comment by Anviking** [comment]
IMO `RecentEra` would still deserve some improved docs.

@HeinrichApfelmus do you think we could agree on some variation of this without too much difficulty?

This example in particular is currently in conflict with the decision
```
myCardanoApiTx :: IsRecentEra era -> RecentEra -> CardanoApi.Tx (CardanoApiEra era)
```
had concerns about it, don't remember if resolved, but can ignore for now, could copy the examples from the decision doc exactly as a first step.

File: lib/balance-tx/lib/internal/Internal/Cardano/Write/Eras.hs [hunk 1]
```diff
@@ -110,13 +110,37 @@ type LatestLedgerEra = Conway
...
+--
+-- === 'RecentEra' the value vs 'IsRecentEra' the class
+--
+-- DO:
+--
+-- @@
+-- myTx :: IsRecentEra era => Tx era
+-- signByAlice :: IsRecentEra era => Tx era -> Tx era
+-- renderCliTx :: IsRecentEra era -> Tx era -> Text
+-- myCardanoApiTx :: IsRecentEra era -> RecentEra -> CardanoApi.Tx (CardanoApiEra era)
```
Comment (HeinrichApfelmus) [inline lib/balance-tx/lib/internal/Internal/Cardano/Write/Eras.hs]:
```suggestion
-- renderCliTx :: IsRecentEra era => Tx era -> Text
-- myCardanoApiTx :: IsRecentEra era => RecentEra era → CardanoApi.Tx (CardanoApiEra era)
```

The other variants do not type check, I believe.

**Comment by HeinrichApfelmus** [comment]
> This example in particular is currently in conflict with the decision
>
> myCardanoApiTx :: IsRecentEra era -> RecentEra -> CardanoApi.Tx (CardanoApiEra era)
>
> had concerns about it, don't remember if resolved,

I think the specific trouble with `myCardanoApiTx` might have been that `CardanoApiEra` is not an injective type family, and specifying the return type is not enough to determine `era` — this type would be ambiguous.

In general, however, the decision document does imply that `era` in the return value is sufficient cause to not have have it as an explicit argument. For example,

```hs
parseThisRecentEra :: IsRecent era => ByteString → Maybe (RecentEra era)
```

is preferred over

```hs
parseThisRecentEra' :: IsRecent era => RecentEra era → ByteString → Maybe (RecentEra era)
```

>  but can ignore for now, could copy the examples from the decision doc exactly as a first step.

Yes, please. Copying the examples from the decision document exactly is a good idea.

**Comment by Anviking** [comment]
@HeinrichApfelmus I've updated the `'RecentEra' the value vs 'IsRecentEra' the class` section to exactly mirror the advice process decision https://github.com/cardano-foundation/cardano-wallet/pull/4296/commits/ad66d40dfedfee382f58ed9a150840700327babb

**Review by HeinrichApfelmus — APPROVED** [review]
Thanks! 😊 👍

**Pull request merged**